tests/integration/cachedfail.py

```diff
@@ -514,3 +425,61 @@ def generate_target():
 
     # Assert that it's no longer cached, and returns to a buildable state
     assert cli.get_element_state(project, "target.bst") == "buildable"
+
+
+# Test that we do not keep scheduling builds after one build fails
+# with `--builders 1` and `--on-error quit`.
+#
+# Note that this depends on staging order (which is stable and a part of cache key calculation),
```

Comment:
It seems that CI disagrees with this statement :) I also think that the order isn't deterministic, given the scheduler queues and threads and whatnot.

The way I was thinking about this is to have two failing builds and assert that one is "failed" and one is "buildable" (not necessarily in this order)

Reply:
Ok, I've taken your approach of just launching two failing builds, and also verified that the new test also fails with your patch reverted... this one should be good !